src/geoext/algorithms.c

```diff
@@ -132,6 +132,33 @@ int point_in_polygon(struct coord2d *pt,
     yflag0 = yflag1;
     vtx0 = vtx1;
   }
-
+  
   return inside_flag;
 }
+
+double area(struct coord2d *coord, int npts)
```

Comment:
this routine is not correct because we have defined that the first and last points of a polygon are the same.
the first loop iteration is performing an evaluation of an expression that will result in zero.